From: Gordon Ball <gordon@chronitis.net>
Date: Mon, 8 Feb 2021 16:13:36 +0100
Subject: Javascript not detected hint

Presumably due to changes in jupyter-client/notebook, we commonly get
"Widget Javascript not found" warnings on starting a kernel. This is due
to the javascript extension loading before the python part does. The
frontend then tries to validate the version and gets no response, but
works in any case once the python library is loaded.

Add a hint to the message that reconnecting to the kernel triggers a
revalidation of the version. Removing the warning completely seems a bit
drastic.
---
 ipywidgets/widgets/widget.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ipywidgets/widgets/widget.py b/ipywidgets/widgets/widget.py
index a23e887..bf1edc7 100644
--- a/ipywidgets/widgets/widget.py
+++ b/ipywidgets/widgets/widget.py
@@ -502,7 +502,8 @@ class Widget(LoggingConfigurable):
             # widget front-end is what is expected.
             if validated is None:
                 loud_error('Widget Javascript not detected.  It may not be '
-                           'installed or enabled properly.')
+                           'installed or enabled properly. Reconnecting the '
+                           'current kernel may help.')
             elif not validated:
                 msg = ('The installed widget Javascript is the wrong version.'
                       ' It must satisfy the semver range %s.'%__frontend_version__)
